python/ray/experimental/channel/serialization_context.py

```diff
@@ -18,10 +18,29 @@ def __init__(self):
         # The number of readers for each channel. When the number of readers
         # reaches 0, remove the data from the buffer.
         self.channel_id_to_num_readers: Dict[str, int] = {}
+        # The key is the order of the task on the actor, which follows the ascending
+        # order of bind_index, and the value is a dictionary of intermediate results.
+        # The dictionary maps the operation type, such as "READ", "COMPUTE", and "WRITE"
+        # , to the intermediate result.
```

Comment:
```suggestion
        # The key is the order of the task on the actor, which follows the ascending
        # order of bind_index, and the value is a dictionary of intermediate results.
        # The dictionary maps the operation type, such as "READ", "COMPUTE", and "WRITE"
        # , to the intermediate result.
```
Use a nested list instead of nested dictionary so we don't have to hash to look up.

Can we reuse the existing "local results" buffer instead?

Also, why is there a return value for the WRITE?

Reply:
Updated https://github.com/ray-project/ray/pull/46911/commits/474e85232248da87566a811a585dd000a10093a1

Reply:
> Use a nested list instead of nested dictionary so we don't have to hash to look up.

After rethinking it, I believe it is impossible for both READ and COMPUTE intermediate results to be in the buffers at the same time.

> Can we reuse the existing "local results" buffer instead?

The above commit uses the existing buffer.

> Also, why is there a return value for the WRITE?

Good catch.